### PR TITLE
feat: return also virtual fields when using accessibleFieldsPlugin

### LINKED ISSUE #739: casl/mongoose - getFields() should also get the virtual properties
**Is your feature request related to a problem? Please describe.**
When using mongoose with the accessibleFieldsPlugin(), the virtual properties are not available. Currently i'm registering the plugin on the models instead of globally, but i have to repeat this on every model.

**Describe the solution you'd like**
Can something like this be implemented as default on the getFields() function?
```javascript
const accessibleFields = accessibleFieldsPlugin(schema, {
    getFields(schema) {
        const paths = Object.keys(schema.paths)
        const virtuals = Object.keys(schema.virtuals)
        return [...paths, ...virtuals]
    },
})
```

### PR DESCRIPTION
**Pull request by jsanchezba**
Fixes #739

### CODE REVIEW

File: packages/casl-mongoose/src/accessible_fields.ts [hunk 1]
```diff
@@ -8,7 +8,10 @@ export type AccessibleFieldsOptions =
   } &
   ({ only: string | string[] } | { except: string | string[] });
 
-export const getSchemaPaths: AccessibleFieldsOptions['getFields'] = schema => Object.keys((schema as { paths: object }).paths);
+export const getSchemaPaths: AccessibleFieldsOptions['getFields'] = schema => [
+  ...Object.keys((schema as { paths: object }).paths),
+  ...Object.keys((schema as { virtuals: object }).virtuals)
+];
```
Comment (stalniy) [inline packages/casl-mongoose/src/accessible_fields.ts:14]: **issue(blocking):** this is a breaking change. And I don't think that everybody needs/wants virtual fields to appear in accessibleFields, especially that some of them may be non-writab

**Comment by stalniy** [comment]
after a 2nd look at the code I don't understand why you can't apply plugin globally with custom `getFields` option, did I miss smth?

```ts
mongoose.plugin(accessibleFieldsPlugin, {
  getFields: schema => [...Object.keys(schema.paths), ...Object.keys(schema.virtuals)]
});
```

File: packages/casl-mongoose/src/accessible_fields.ts (hunk 1 above)
Comment (jsanchezba) [reply packages/casl-mongoose/src/accessible_fields.ts:14]: > after a 2nd look at the code I don't understand why you can't apply plugin globally with custom `getFields` option, did I miss smth?
> 
> ```ts
> mongoose.plugin(accessibleFieldsPlugin, {
>   getFields: schema => [...Object.keys(schema.paths), ...Object.keys(schema.virtuals)]
> });
> ```

After looking at it again i was able to make it work globally, i don't know why it didn't work the first time, guess i did something wrong. It's a bit confusing some times when working with typescript.

And yeah, it's a breaking change and if it works globally with a few lines i don't think it's worth it. I'll better close this PR.
It was all my misunderstanding, sorry

**Comment by jsanchezba** [comment]
> after a 2nd look at the code I don't understand why you can't apply plugin globally with custom `getFields` option, did I miss smth?
> 
> ```ts
> mongoose.plugin(accessibleFieldsPlugin, {
>   getFields: schema => [...Object.keys(schema.paths), ...Object.keys(schema.virtuals)]
> });
> ```

After looking at it again i was able to make it work globally, i don't know why it didn't work the first time, guess i did something wrong. It's a bit confusing some times when working with typescript.

And yeah, it's a breaking change and if it works globally with a few lines i don't think it's worth it. I'll better close this PR.
It was all my misunderstanding, sorry

**Pull request closed without merging**